pkg/ddevapp/ddevapp.go

```diff
@@ -46,87 +46,106 @@ const SiteConfigMissing = ".ddev/config.yaml missing"
 // SiteStopped defines the string used to denote when a site is in the stopped state.
 const SiteStopped = "stopped"
 
-// DdevApp is the struct that represents a ddev app
+// DdevApp is the struct that represents a ddev app, mostly its config
+// from config.yaml.
 type DdevApp struct {
-	AppConfig *Config
+	APIVersion            string               `yaml:"APIVersion"`
+	Name                  string               `yaml:"name"`
+	AppType               string               `yaml:"type"`
+	Docroot               string               `yaml:"docroot"`
+	WebImage              string               `yaml:"webimage"`
+	DBImage               string               `yaml:"dbimage"`
+	DBAImage              string               `yaml:"dbaimage"`
+	ConfigPath            string               `yaml:"-"`
+	AppRoot               string               `yaml:"-"`
+	Platform              string               `yaml:"-"`
+	Provider              string               `yaml:"provider,omitempty"`
+	DataDir               string               `yaml:"-"`
+	ImportDir             string               `yaml:"-"`
+	SiteSettingsPath      string               `yaml:"-"`
+	SiteLocalSettingsPath string               `yaml:"-"`
+	providerInstance      Provider             `yaml:"-"`
+	Commands              map[string][]Command `yaml:"hooks,omitempty"`
 }
 
 // GetType returns the application type as a (lowercase) string
-func (l *DdevApp) GetType() string {
-	return strings.ToLower(l.AppConfig.AppType)
+func (app *DdevApp) GetType() string {
+	return strings.ToLower(app.AppType)
 }
 
 // Init populates DdevApp config based on the current working directory.
 // It does not start the containers.
-func (l *DdevApp) Init(basePath string) error {
-	config, err := NewConfig(basePath, "")
-
-	// Save config to l.AppConfig so we can capture and display the site's
-	// status regardless of its validity
-	l.AppConfig = config
-
+func (app *DdevApp) Init(basePath string) error {
+	newApp, err := NewApp(basePath, "")
 	if err != nil {
 		return err
 	}
 
-	err = config.Validate()
+	err = newApp.ValidateConfig()
 	if err != nil {
 		return err
 	}
 
-	web, err := l.FindContainerByType("web")
+	*app = *newApp
+	web, err := app.FindContainerByType("web")
+
+	// if err == nil, it means we have found some containers. Make sure they have
+	// the right stuff in them.
 	if err == nil {
 		containerApproot := web.Labels["com.ddev.approot"]
-		if containerApproot != l.AppConfig.AppRoot {
-			return fmt.Errorf("a web container in %s state already exists for %s that was created at %s", web.State, l.AppConfig.Name, containerApproot)
+		if containerApproot != app.AppRoot {
+			return fmt.Errorf("a web container in %s state already exists for %s that was created at %s", web.State, app.Name, containerApproot)
 		}
+		return nil
+	} else if strings.Contains(err.Error(), "could not find containers") {
```

Comment:
maybe a quick note why this error is ok to hit. took me a sec to remember the recent changes for list/describe broken projects.

Reply:
Added.